test/test_utils/ec2.py

```diff
@@ -693,14 +704,48 @@ def execute_ec2_habana_training_performance_test(
     cap_add = '--cap-add=sys_nice'
     ipc = '--ipc=host' if "pytorch" in ecr_uri else ""
     habana_container_test_repo = '${HOME}/gaudi-test-suite:/gaudi-test-suite'
-    connection.run(
-        f"{docker_cmd} run --user root "
-        f"-e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} "
-        f"-e PR_CONTEXT={1 if is_pr_context() else 0} "
-        f"{container_runtime} {ompi_mca_btl} {hpu_env_vars} {cap_add} {ipc} "
-        f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} -v {habana_container_test_repo} "
+    execution_command = f"{docker_cmd} run --user root " \
+        f"-e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} " \
+        f"-e PR_CONTEXT={1 if is_pr_context() else 0} " \
+        f"{container_runtime} {ompi_mca_btl} {hpu_env_vars} {cap_add} {ipc} " \
+        f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} -v {habana_container_test_repo} " \
         f"{ecr_uri} {os.path.join(os.sep, 'bin', 'bash')} -c '{test_cmd}'"
+    
+    framework = "tensorflow" if "tensorflow" in ecr_uri else "pytorch" if "pytorch" in ecr_uri else None
+    account_id_prefix = os.getenv("ACCOUNT_ID", boto3.client("sts").get_caller_identity()["Account"])[:3]
+    s3_bucket_for_permanent_logs = f"dlinfra-habana-tests-{account_id_prefix}"
+    test_type = "benchmark"
+    custom_filename = test_cmd.split(f"{os.sep}")[-1]
+    custom_filename += f"cards-{cards_num}" if cards_num else f"cards-0"
+    s3_uri_permanent_logs = get_s3_uri_for_saving_permanent_logs(
+        framework, s3_bucket=s3_bucket_for_permanent_logs, test_type=test_type, custom_filename=custom_filename
+    )
+    LOGGER.info(f"******** Going for Async Execution for {test_cmd} ********")
+    required_log_ending = "Kudos!! Habana tests executed successfully"
```

Comment:
Shouldn't it be after `execute_asynchronus_testing_using_s3_bucket`?

Reply:
No, it is just a variable that is sent to `execute_asynchronus_testing_using_s3_bucket`. `execute_asynchronus_testing_using_s3_bucket` takes care of everything.